tablo/csv_utils.py

```diff
@@ -38,15 +42,37 @@ def prepare_csv_rows(csv_file):
 
     DateType.formats = create_date_formats(day_first=False)
 
-    # We are never wanting boolean types, so remove that from the default list
-    eligible_types = [StringType, DecimalType, IntegerType, DateType]
-    types = type_guess(row_set.sample, types=eligible_types, strict=True)
+    if csv_info:
+        # If we have csv_info, then we know the column types, no need to guess
+        types = apply_csv_types(row_set.sample, csv_info)
+    else:
+        types = type_guess(row_set.sample, strict=True)
 
     row_set.register_processor(types_processor(types))
 
     return row_set
 
 
+def apply_csv_types(rows, csv_info):
```

Comment:
might be better as `types_from_config` since you are passing in a config (`csv_info`) and making that into types.